roles/openshift_metrics/tasks/generate_cassandra_pvcs.yaml

```diff
@@ -1,17 +1,23 @@
 ---
-- name: Check to see if PVC already exists
+- name: Check to see if PVC already exists in actual namespace
   oc_obj:
     state: list
     kind: pvc
     name: "{{ openshift_metrics_cassandra_pvc_prefix }}-{{ metrics_pvc_index }}"
-    namespace: "{{ openshift_metrics_project }}"
-  register: _metrics_pvc
+    namespace: "{{openshift_metrics_project}}"
+  register: metrics_pvc
+
+ # Migration process for PV to another namespace.
+- name: Check if we need to move cassandra PV to another namespace.
+  include_tasks: migrate_cassandra_pvcs.yaml
 
 # _metrics_pvc.results.results | length > 0 returns a false positive
 # so we check for the presence of 'stderr' to determine if the obj exists or not
 # the RC for existing and not existing is both 0
+
+# No PVC exist (either new or old namespace), so it's a new installation
 - when:
-    - _metrics_pvc.results.stderr is defined
+    - metrics_pvc.results.stderr is defined and not openshift_metrics_migrate_namespace
```

Comment:
I think this check is good. We proceed in this block only if neither the PVC exists in the new namespace nor in the old namespace.